### PR TITLE
IMDB NLP, Fashion-MNIST, MNIST, Mushroom Classification Demos Completed

### PR DESCRIPTION
**Pull request by sram-alectio**

### CODE REVIEW

File: alectio_sdk/flask_wrapper/pipeline.py [hunk 1]
```diff
@@ -142,7 +142,7 @@ def _one_loop(self, payload, args):
             object_key = os.path.join(self.expt_dir, "data_map.pkl")
             self.client.multi_part_upload_with_s3(self.state_json, self.bucket_name, object_key, "pickle")
         else:
-            self.resume_from = "ckpt_{}".format(self.cur_loop - 1)
+            self.resume_from = "ckpt_{}".format(self.cur_loop)
```
Comment (arunsanknar) [inline alectio_sdk/flask_wrapper/pipeline.py]: No changes should ever be made here, its completely restricted to change this code. The only code blocks that should be added and tested is within the examples section

File: alectio_sdk/flask_wrapper/config.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
   "sandbox_bucket": "",
-  "backend_ip": "http://ad9b8064978b349c2aefcbb289e27b90-1345512929.us-west-2.elb.amazonaws.com/",
+  "backend_ip": "a9c837bf7b825453ca80be71d4c39a22-1823745019.us-west-2.elb.amazonaws.com",
   "backend_route": "127.0.0.1:5000/end_of_task"
```
Comment (arunsanknar) [inline alectio_sdk/flask_wrapper/config.json]: The latest changes on develop don't contain this. Pull the latest changes from the base branch for the same.

File: examples/mushroom_classification/main.py [hunk 3]
```diff
@@ -21,6 +21,6 @@
 )
 
 if __name__ == "__main__":
-    # payload = json.load(open(args["sample_payload"], "r"))
-    # app._one_loop(args=args, payload=payload)
-    app(debug=True)
+    payload = json.load(open(args["sample_payload"], "r"))
+    app._one_loop(args=args, payload=payload)
+    # app(debug=True)
```
Comment (arunsanknar) [inline examples/mushroom_classification/main.py]: This can only be used for testing purposes on your local and this should be removed when you do an end to end testing

File: examples/mushroom_classification/processes.py [hunk 4]
```diff
@@ -19,14 +9,43 @@
...
 import torch.optim as optim
 import os
+import yaml
+import argparse
 
 device = "cuda" if torch.cuda.is_available() else "cpu"
 
-def getdatasetstate(args={}):
-    return {k: k for k in range(120000)}
+print("using device", device)
```
Comment (arunsanknar) [inline examples/mushroom_classification/processes.py]: Remove these print statements. Refer to other examples so that they remain consistent with other examples on the codebase

File: examples/mushroom_classification/processes.py [hunk 5]
```diff
@@ -19,14 +9,43 @@
...
+
+def getdatasetstate(args, split="train"):
+    global train_dataset, test_dataset
+    
+    CSV_FILE = "./data/datasets_478_974_mushrooms.csv"
+    dataset = MushroomDataset(CSV_FILE)
+
+    train_test = torch.utils.data.random_split(dataset, (int(0.8*len(dataset)), len(dataset)-int(0.8*len(dataset))))
```
Comment (arunsanknar) [inline examples/mushroom_classification/processes.py]: seed these splits. It cannot be random. Instead of splitting the dataset here ,rather use another python file to do the preprocessing/splitting. The getdatasetstate function block should only be used to get the state of an already existing dataset and not to split and preprocess the data

**Comment by arunsanknar** [comment]
The PR should follow the new PR submission guidelines. When you raise a PR there will be a predefined template. Fill out those templates and answer the questions and raise another PR. Also the main modules of the SDK cannot be changed. Make the changes from the initial review, retest them and raise another PR
